R/streamable_table.R

```diff
@@ -202,16 +200,79 @@ streamable_base_csv <- function() {
   ## NOTE: write.csv does not permit setting 
   ## `col.names = FALSE``, so cannot omit_header
   write_csv <- function(x, path, omit_header) {
-      utils::write.table(x,
+    utils::write.table(x,
                        file = path, 
                        sep = ",", 
                        quote = TRUE,
                        qmethod = "double",
                        row.names = FALSE,
                        col.names = !omit_header,
                        append = omit_header
-      )
+    )
   }
   streamable_table(read_csv, write_csv, "csv")
 }
 
+
+#' streamable chunked parquet using `arrow`
+#' 
+#' @return a `streamable_table` object (S3)
+#' @export
+#' @seealso [arrow::read_parquet()], [arrow::write_parquet()]
+streamable_parquet <- function() {
+  
+  ## Avoids a hard dependency on arrow for this courtesy function
+  if (!requireNamespace("arrow", quietly = TRUE)) {
+    stop("arrow package must be installed to use parquet",
+         call. = FALSE)
+  }
+  
+  read_parquet <- getExportedValue("arrow", "read_parquet")
+  write_parquet <- getExportedValue("arrow", "write_parquet")
+  
+  read <- function(file, ...) {
+    read_parquet(file, ...)
+  }
+  
+  write <- function(x, path, omit_header = FALSE, ...) {
+    # 1. Store parquet pieces in a directory, create the directory.
+    # 2. Get the number of the last piece via list.files.
+    # 3. Increment the part number, rewrite path, write part to disk.
+    # 4. Profit
+    
+    # Store the parquet pieces in a directory named after the table 
+    # for ease of use with arrow::open_dataset
+    dir_path <- paste0(
+      dirname(path), 
+      "/",
+      strsplit(
```

Comment:
When https://github.com/ropensci/arkdb/pull/40/files#r715883060 is done, this strsplit can be removed.